src/lib/tradingView/dydxfeed/utils.ts

```diff
@@ -0,0 +1,106 @@
+import BigNumber from 'bignumber.js';
+import { sum } from 'lodash';
+import { Mark, MarkCustomColor, ResolutionString } from 'public/tradingview/charting_library';
+
+import { AbacusOrderSide, SubaccountFills } from '@/constants/abacus';
+import { RESOLUTION_TO_INTERVAL_MS } from '@/constants/candles';
+import { STRING_KEYS, StringGetterFunction, SupportedLocales } from '@/constants/localization';
+import { TOKEN_DECIMALS } from '@/constants/numbers';
+import { ThemeColorBase } from '@/constants/styles/colors';
+
+import { formatNumberOutput, OutputType } from '@/components/Output';
+
+import { RootStore } from '@/state/_store';
+import { getFillDetails } from '@/state/accountSelectors';
+
+import { getAverageFillPrice } from '@/lib/orders';
+
+/**
+ * @description Converts times in ms to the appropriate bar time (in seconds)
+ * For example, if the starting time = 5000ms, interval = 10,000ms, a value of 26,000ms would
+ * be grouped into the bar at 25,000ms = 25s
+ */
+export function getBarTime(
+  chartStartTimeMs: number,
+  fillTimeMs: number,
+  resolution: ResolutionString
+): number | undefined {
+  const intervalMs = RESOLUTION_TO_INTERVAL_MS[resolution];
+
+  const [startBn, intervalSizeBn, fillTimeBn] = [
+    BigNumber(chartStartTimeMs),
+    BigNumber(intervalMs),
+    BigNumber(fillTimeMs),
+  ];
+  const numIntervalsBetween = fillTimeBn.minus(startBn).dividedToIntegerBy(intervalSizeBn);
+  return startBn
+    .plus(numIntervalsBetween.multipliedBy(intervalSizeBn))
+    .dividedToIntegerBy(1000)
+    .toNumber();
+}
+
+export const getMarkForOrderFills = (
```

Comment:
[q] i noticed we tested getBarTime but i don't see a test for getMarkForOrderFills in the spec file. is this because the overhead to create all the necessary fixtures is too high? 

if so, could we make a ticket + comment it in code as a TODO. If we find that there's a lot of instances where we're unable to test due to lack of features it may be worth actively investing time to remediate this

Reply:
hm so i personally dont think it's that important to add a unit test here since this is just a function that prepares all the little UI components - probably more interesting to add unit tests separately for each piece (like i did for `getBarTime`)